components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -671,7 +671,14 @@ def canChgrp(self):
         group. Web client will only allow this for the data Owner. Admin CAN
         move other user's data, but we don't support this in Web yet.
         """
-        return self.isOwned() or self._conn.isAdmin()   # See #8974
+        return self.getDetails().getPermissions().canChgrp()
```

Comment:
Guess I'm mostly worried if `canChgrp` ever got used as a proxy for something else that now differs from the server-side logic.